unit/sharing_node.cpp

```diff
@@ -1,3 +1,14 @@
+/*******************************************************************\
+
+Module: Unit tests for sharing node
+
+Author: Daniel Poetzl, daniel.poetzl@cs.ox.ac.uk
+
+\*******************************************************************/
+
+/// \file
```

Comment:
I'm not quite sure what the changes to sharing tree have to do with this PR but ... sure?

Reply:
Unit tests did not compile with -DNDEBUG for the same reason as regular code. After fixing the files, the linter script complained about the missing copyright statement, so I fixed that as well.

Reply:
Could that be placed in a separate commit? Not a big deal, but it seems like a change that should never get reverted even when others would.

Reply:
@tautschnig : you mean the changes to `src/util/sharing_node.h` _and_ `unit/sharing_node.cpp` should be a separate commit, not just the copyright statement in `unit/sharing_node.cpp`. Correct?